### PR TITLE
Remove leading 'v' from version strings (#19456).

### PR DESCRIPTION
**Pull request by wholtz**
Bioconda requires reviews prior to merging pull-requests (PRs). To facilitate this, once your PR is passing tests and ready to be merged, please add the `please review & merge` label so other members of the bioconda community can have a look at your PR and either make suggestions or merge it.

* [x] I have read the [guidelines for bioconda recipes](https://bioconda.github.io/contributor/guidelines.html).
* [ ] This PR adds a new recipe.
* [x] AFAIK, this recipe **is directly relevant to the biological sciences** (otherwise, please submit to the more general purpose [conda-forge channel](https://conda-forge.org/docs/)).
* [x] This PR updates an existing recipe.
* [ ] This PR does something else (explain below).

### CODE REVIEW

**Comment by dpryan79** [comment]
@bioconda-bot please fetch artifacts

**Comment by bgruening** [comment]
@wholtz cool. Thanks a lot!

We just need to decide what happens to the old builds. Do we get the non-v version by default after merging this?

**Comment by dpryan79** [comment]
Hmm, I fear that we'll get the v builds, since v is > any number in ascii.

@mbargull Do you happen to know if conda has any built in logic to exclude `v` prefixes in version numbers?

**Comment by mbargull** [comment]
The fixed versions will all be considered newer by `conda` since any letter-based portions are handled like "`a` for `alpha`" & co.
Meaning, e.g., `0.0.1a1 < 0.0.1b1 < 0.0.1`, and any missing number before the letters is assumed to be zero, e.g., `v0.0.1 = 0v0.0.1 < 0.0.1`.

File: recipes/perl-filedirutil/meta.yaml [hunk 1]
```diff
@@ -11,7 +11,7 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 0
+  number: 1
```
Comment (mbargull) [inline recipes/perl-filedirutil/meta.yaml]:
```suggestion
  number: 0
```

File: recipes/perl-bio-viennangs/meta.yaml [hunk 2]
```diff
@@ -11,7 +11,7 @@ source:
   sha256: {{ sha256 }}
 
 build:
-  number: 5
+  number: 6
```
Comment (mbargull) [inline recipes/perl-bio-viennangs/meta.yaml]:
```suggestion
  number: 0
```

File: recipes/perl-bio-rna-rnaalisplit/meta.yaml [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
 
 source:
-  url: https://cpan.metacpan.org/authors/id/M/MT/MTW/Bio-RNA-RNAaliSplit-{{ version }}.tar.gz
+  url: https://cpan.metacpan.org/authors/id/M/MT/MTW/Bio-RNA-RNAaliSplit-v{{ version }}.tar.gz
   sha256: {{ sha256 }}
 
 build:
   noarch: generic
-  number: 1
+  number: 2
```
Comment (mbargull) [inline recipes/perl-bio-rna-rnaalisplit/meta.yaml]:
```suggestion
  number: 0
```

**Comment by wholtz** [comment]
I tried to fix all the version strings but two of the builds failed. Therefore, in 9ffd8aa I reverted the changes to the recipes for deploid and rapidnj.

**Comment by mbargull** [comment]
Yep, I noticed 9ffd8aae3f616c4bc1ae1760ed2eec546e0cf87a, and we should definitely fix those too. But you don't have to dig into their build failures right here (unless you're overly motivated and want to, of course ;)). It's fine to just open an issue for us to track this so we fix this at some point.

Thanks for your initiative in any case!

**Comment by dpryan79** [comment]
Great, thanks for clarifying things @mbargull :)

Let's tweak the build numbers and merge this then!

**Comment by dpryan79** [comment]
@bioconda-bot please merge

**Pull request merged**